File: qlty-cli/tests/lang/java/basic.in/Lines.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

class Main {
    public static void f1() {
        try {
            File sourceFile = File.createTempFile("ruby", ".kt");
            try (FileWriter writer = new FileWriter(sourceFile)) {
                writer.write("foo(...args)");
            }

            Object tree = parseFile(sourceFile);

            bar();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static Object parseFile(File file) {
        return new Object();
    }

    // Foo
    public static void f2() {
        bar(); // does not count as comment line
    }

    // multi-line comment
    /*
     * line1
     * line2
     * line4
     */

    public static void f3() {
        bar();
    }

    public static void bar() {
        System.out.println("bar() called");
    }
}

File: qlty-cli/tests/lang/csharp/basic.in/Lines.cs
using System;
using System.IO;

public class Main
{
    public static void F1()
    {
        try
        {
            string tempFilePath = Path.Combine(
                Path.GetTempPath(),
                $"ruby{Guid.NewGuid()}.kt"
            );
            
            using (var writer = new StreamWriter(tempFilePath))
            {
                writer.Write("foo(...args)");
            }

            object tree = ParseFile(tempFilePath);

            Bar();
        }
        catch (IOException e)
        {
            Console.Error.WriteLine(e);
        }
    }

    public static object ParseFile(string filePath)
    {
        return new object();
    }

    // Foo
    public static void F2()
    {
        Bar(); // does not count as comment line
    }

    // multi-line comment
    /*
     * line1
     * line2
     * line4
     */

    public static void F3()
    {
        Bar();
    }

    public static void Bar()
    {
        Console.WriteLine("bar() called");
    }
}
